src/lib/SpecialFileDirGen.cc

```diff
@@ -4,6 +4,58 @@
 
 namespace simeng {
 
+// Wrapper around calls to "system(command)". Checks that a shell is available
+// before calling "system" and checking the output for any issues.
+// ensureExitSuccess is used to check for a successful termination status (0)
+// from the child shell, defaults to on
+int systemWrapper(const std::string& command,
+                  const bool ensureExitSuccess = true) {
+  // Check that there is a shell available
+  if (!system(NULL)) {
+    std::cerr
+        << "[SimEng:SpecialFileDirGen] Shell unavailable, can't call system"
+        << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  int output = system(command.c_str());
+
+  if (output == -1) {
+    std::cerr << "[SimEng:SpecialFileDirGen] Child process could not be "
+                 "created, or its status could "
+                 "not be retrieved. errno = "
+              << errno << std::endl;
+    exit(EXIT_FAILURE);
+  } else if (WIFEXITED(output) && WEXITSTATUS(output) == 127) {
+    std::cerr << "[SimEng:SpecialFileDirGen] Shell command could not be "
+                 "executed in child shell"
+              << std::endl;
+    exit(EXIT_FAILURE);
+  } else {
+    if (ensureExitSuccess) {
+      if (WIFEXITED(output) && WEXITSTATUS(output) == 0) {
```

Comment:
It is the termination status of the shell (which is the termination status of the last command it ran) which SHOULD be 0 on success. I guess it could be anything at all depending on what the command being executed does. If it does go wrong, the catch all error message should be a very good start for debugging. See man pages in comment https://github.com/UoB-HPC/SimEng/pull/408#discussion_r1560930125 and also look at waitpid